apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/UploadSurveyBg.tsx

```diff
@@ -0,0 +1,95 @@
+import { debounce } from "lodash";
+import { Loader, SearchIcon } from "lucide-react";
+import Image from "next/image";
+import { useEffect, useRef, useState } from "react";
+
+import { env } from "@formbricks/lib/env";
+import { Input } from "@formbricks/ui/Input";
+
+interface UploadSurveyBgProps {
+  handleBgChange: (url: string, bgType: string) => void;
+  background: string;
+}
+
+interface Image {
+  id: string;
+  alt_description: string;
+  urls: {
+    regular: string;
+  };
+}
+
+export const UploadSurveyBg = ({ handleBgChange, background }: UploadSurveyBgProps) => {
+  const inputFocus = useRef<HTMLInputElement>(null);
+  const [loading, setLoading] = useState(false);
+  const [query, setQuery] = useState("");
+  const [images, setImages] = useState<Image[]>([]);
+
+  useEffect(() => {
+    const fetchData = async (searchQuery: string) => {
+      setLoading(true);
+      try {
+        const response = await fetch(
+          `https://api.unsplash.com/search/photos?query=${searchQuery}&client_id=${env.NEXT_PUBLIC_UNSPLASH_API_KEY}&orientation=landscape&w=1920&h=1080`
+        );
+        const data = await response.json();
+        setImages(data.results);
+        setLoading(false);
+      } catch (error) {
+        setLoading(false);
+      }
+    };
+
+    const debouncedFetchData = debounce(fetchData, 500);
+
+    if (query.trim() !== "") {
+      debouncedFetchData(query);
+    }
+
+    return () => {
+      debouncedFetchData.cancel();
+    };
+  }, [query, setImages]);
+
+  useEffect(() => {
+    inputFocus.current?.focus();
+  }, []);
```

Comment:
Can you also try to reduce the number of useEffect calls if possible?

Reply:
This useEffect run only once when the component mounts. And another useEffect runs when query and setImage changes. So it's good to seperate both the useEffect.